tests/e2e/tests/pilot/routing_test.go

```diff
@@ -653,3 +653,119 @@ func TestHeadersManipulations(t *testing.T) {
 		})
 	}
 }
+
+func TestDestinationRuleConfigScope(t *testing.T) {
+	samples := 100
+
+	var cfgs *deployableConfig
+	applyRuleFunc := func(t *testing.T, ns, ruleYaml string) {
+		// Delete the previous rule if there was one. No delay on the teardown, since we're going to apply
+		// a delay when we push the new config.
+		if cfgs != nil {
+			if err := cfgs.TeardownNoDelay(); err != nil {
+				t.Fatal(err)
+			}
+			cfgs = nil
+		}
+
+		// Apply the new rule
+		cfgs = &deployableConfig{
+			Namespace:  ns,
+			YamlFiles:  []string{ruleYaml},
+			kubeconfig: tc.Kube.KubeConfig,
+		}
+		if err := cfgs.Setup(); err != nil {
+			t.Fatal(err)
+		}
+	}
+	// Upon function exit, delete the active rule.
+	defer func() {
+		if cfgs != nil {
+			_ = cfgs.Teardown()
+		}
+	}()
+
+	cases := []struct {
+		testName        string
+		description     string
+		config          string
+		namespace       string
+		scheme          string
+		src             string
+		dst             string
+		expectedSuccess bool
+		operation       string
+		onFailure       func()
+	}{
+		{
+			testName:        "private destination rule in same namesapce",
+			description:     "routing all traffic to c-v1",
+			config:          "destination-rule-c-private.yaml",
+			namespace:       tc.Kube.Namespace,
+			scheme:          "http",
+			src:             "a",
+			dst:             "c",
+			expectedSuccess: true,
+			operation:       "c.istio-system.svc.cluster.local:80/*",
+		},
+		{
+			testName:        "private destination rule in different namespaces",
+			description:     "can not apply private destination rule",
+			config:          "destination-rule-c-private.yaml",
+			namespace:       "test-diff-ns",
+			scheme:          "http",
+			src:             "a",
+			dst:             "c",
+			expectedSuccess: false,
+			operation:       "c.istio-system.svc.cluster.local:80/*",
+		},
```

Comment:
also a test for private destination rule in a namespace overrides public destination rule from another namespace.

Reply:
done